pulseapi/entries/models.py

```diff
@@ -68,6 +68,12 @@ def with_related(self):
             'related_entry_creators__profile__related_user',
         )
 
+    def by_active_profile(self):
+        """
+        Return all entries that have been created by pulse users who's profiles are set to "Active" by moderator.
+        """
+        return self.filter(published_by__profile__is_active=True)
+
```

Comment:
**Question** There already is the `public` method on the `EntryQuerySet`. From the docstring and the filter it appears to serve the same purpose (only show moderator approved content). It seems like anything that we would want to show publically should fulfill both, the "approved" moderation state and that the user profile is "active". Is there need to split these checks? Or could we include the "profile is active" check in the `public` method?

Reply:
Hi Tibor!

The public() filter is used for two querysets: the `EntryQuerySet` which is used to show individual entries and the `EntryListQuerySet` which is used to display a list of all public entries by approved profiles.

The EntryQuerySet is called when a user clicks on an entry from their profile, regardless of whether they are set to `is_active` or not. If we were to combine these two filter into one, these inactive users would no longer be able to visit their posts until they are set to active which is not the desired workflow.

Let me know if that helps, thanks!

Reply:
That makes sense. Thanks @danielfmiranda. If this is for users to see their own posts, we could potentially also allow them to see posts that are not approved. Or am I missing something?